### PR TITLE
feat(virtiofs): virtiofs root filesystem support

### PR DESCRIPTION
**Pull request by germag**
Currently, dracut can't mount virtiofs as root filesystem.
Make possible to boot virtual machines off virtiofs,
passing "root=virtiofs:\<mtag\>", using the mount tag <mtag>
as root filesystem. Alternatively this module also supports
"rootfstype=virtiofs root=\<mtag\>"

## Checklist
- [ x ] I have tested it locally
- [ ] I have reviewed and updated any documentation if relevant
- [ ] I am providing new code and test(s) for it

### CODE REVIEW

File: modules.d/95virtiofs/mount-virtiofs.sh [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+virtiofs_fstype_opt() {
+	[ -n "$fstype" -a "$fstype" = "virtiofs" ]
+}
+
+virtiofs_root_opt() {
+	[ -n "$root" -a -z "${root%%virtiofs:*}" ]
+}
+
+virtiofs_root() {
+	[ virtiofs_fstype_opt -o virtiofs_root_opt ]
```
Comment (dtardon) [inline modules.d/95virtiofs/mount-virtiofs.sh:12]: This line doesn't do what you think it does. It's the equivalent of 
```sh
[ -n 'virtiofs_fstype_opt' -o -n 'virtiofs_root_opt' ]
```
You probably meant to write
```sh
virtiofs_fstype_opt || virtiofs_root_opt
```

File: modules.d/95virtiofs/mount-virtiofs.sh [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+}
+
+load_module() {
+	if ! (modprobe virtiofs || strstr "$(cat /proc/filesystems)" virtiofs); then
```
Comment (dtardon) [inline modules.d/95virtiofs/mount-virtiofs.sh:16]: Please use
```sh
if ! { modprobe virtiofs || strstr "$(cat /proc/filesystems)" virtiofs; }; then
```
instead. There's no need to run a subshell here.

File: modules.d/95virtiofs/mount-virtiofs.sh [hunk 3]
```diff
@@ -0,0 +1,40 @@
+#!/usr/bin/sh
+
+virtiofs_fstype_opt() {
+	[ -n "$fstype" -a "$fstype" = "virtiofs" ]
```
Comment (dtardon) [inline modules.d/95virtiofs/mount-virtiofs.sh:4]: It's not necessary to test `$fstype` for non-emptiness if you then compare it with a concrete string, is it?

File: modules.d/95virtiofs/mount-virtiofs.sh [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+}
+
+virtiofs_root_opt() {
+	[ -n "$root" -a -z "${root%%virtiofs:*}" ]
```
Comment (dtardon) [inline modules.d/95virtiofs/mount-virtiofs.sh:8]: I think that
```sh
[ ${root%%:*} = "virtiofs" ]
```
is a bit easier to understand. But YMMV.

File: modules.d/95virtiofs/mount-virtiofs.sh [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+	fi
+}
+
+mount_root() {
+	local fstag=${root#virtiofs:}
+	mount -t virtiofs -o "$rflags" "$fstag" "$NEWROOT" 2>&1 | vinfo
+}
+
+if virtiofs_root; then
+	load_module
```
Comment (dtardon) [inline modules.d/95virtiofs/mount-virtiofs.sh:27]: Btw, are all these small functions, each only called from a single place, really necessary? The code would be half of the current length without them and IMHO still perfectly understandable...

File: modules.d/95virtiofs/mount-virtiofs.sh [hunk 6]
```diff
@@ -0,0 +1,40 @@
...
+}
+
+load_module() {
+	if ! (modprobe virtiofs || strstr "$(cat /proc/filesystems)" virtiofs); then
+		die "virtiofs is required but not available."
+	fi
+}
+
+mount_root() {
+	local fstag=${root#virtiofs:}
```
Comment (dtardon) [inline modules.d/95virtiofs/mount-virtiofs.sh:22]: IMHO there's no need for a variable here. Just use the expression directly.

File: .github/labeler.yml [hunk 7]
```diff
@@ -228,6 +228,9 @@ udev-rules:
 virtfs:
   - modules.d/95virtfs/*
 
+virtiofs:
```
Comment (dtardon) [inline .github/labeler.yml:231]: Nitpick: I'd split this into a separate commit, as it's an infrastructure thing, not a part of the feature.

**Comment by germag** [comment]
v3:
- Fix shellcheck error
When I ran 'shfmt -s' locally (in v2), it found no issues

**Review by johannbg — APPROVED** [review]
LGTM

Nice touch with the label, Appreciated thanks.
